driver/network_driver.go

```diff
@@ -574,25 +657,31 @@ RETRY_CONTAINER_INSPECT:
 		}
 		labelsFound++
 		labelClean := strings.TrimPrefix(label, DOCKER_LABEL_PREFIX)
-		endpoint.Metadata.Labels[labelClean] = labelValue
+		endpoint.ObjectMeta.Labels[labelClean] = labelValue
 		log.Debugf("Found label for WorkloadEndpoint: %s=%s", labelClean, labelValue)
 	}
 
 	if labelsFound == 0 {
-		log.Debugf("No labels found for container (T=%s)", endpointID, time.Since(start))
+		log.Debugf("No labels found for container (T=%s)", time.Since(start))
 		return
 	}
 
+	rev, err := strconv.Atoi(endpoint.ObjectMeta.ResourceVersion)
```

Comment:
I don't think this is right - could you remove the bit that updates the resource version?

Reply:
@tmjd  however, it will failed when updating with same rev if CALICO_LIBNETWORK_LABEL_ENDPOINT set true.

```ERRO[0086] Unable to update WorkloadEndpoint with labels (T=400.626763ms): update conflict: WorkloadEndpoint(libnetwork/slave-libnetwork-libnetwork-c8f7152c607b86720cf117bf95a3cb3b00c02d2b05215e2859cdab8004735e45)```

thus means we have to update revision.

Reply:
I think that's because when CALICO_LIBNETWORK_LABEL_ENDPOINT is true, there are two goroutines which are attempting to udpate the workload endpoint, and so one of them will fail (because it will have an old revision).

Rather than manually adjust the revision, we should use a retry loop for this case.

We do something along these lines in other similar bits of code:

```
for range(10) {
  wep, err := Get(wep)
  if err != nil {
    break
  }

  err := Update(wep)
  if err != nil && err == UpdateConflict {
    // Conflict updating WEP - retry
    time.Sleep(10ms)
    continue
  }
break
}
```